pos_restricted_customer_list/static/src/js/pos_restricted_customer_list.js

```diff
@@ -0,0 +1,57 @@
+odoo.define("pos_restricted_customer_list.point_of_sale.models", function(require) {
+    "use strict";
+
+    var Model = require("web.DataModel");
+    var PosModels = require("point_of_sale.models");
+    var PosModel = PosModels.PosModel;
+    var PosModelSuper = PosModel.prototype;
+
+    PosModels.PosModel = PosModel.extend({
+        initialize: function(session, attributes) {
+            var self = this;
+            for (var i = 0; i < self.models.length; i++) {
+                var model = self.models[i];
+                var model_name = model.model;
+
+                if (model_name === "res.partner") {
+                    model.domain = self.prepare_load_new_partners_domain();
+                }
+            }
+            return PosModelSuper.initialize.call(self, session, attributes);
+        },
+        prepare_load_new_partners_domain: function() {
+            return [
+                ["customer", "=", true],
+                ["available_in_pos", "=", true],
+            ];
+        },
+        load_new_partners: function() {
+            var self = this;
+            var def = new $.Deferred();
+            var fields = _.find(this.models, function(model) {
+                return model.model === "res.partner";
+            }).fields;
+            var domain = self.prepare_load_new_partners_domain();
+            new Model("res.partner")
+                .query(fields)
+                .filter(domain)
+                .all({timeout: 3000, shadow: true})
+                .then(
+                    function(partners) {
+                        if (self.db.add_partners(partners)) {
+                            // Check if the partners we got were real updates
+                            def.resolve();
+                        } else {
+                            def.reject();
+                        }
+                    },
+                    // eslint-disable-next-line handle-callback-err
+                    function(err, event) {
+                        event.preventDefault();
+                        def.reject();
+                    }
+                );
+            return def;
+        },
+    });
+});
```

Comment:
I think this is not ok.

I can't find a reason to overload `load_new_partners`:
https://github.com/odoo/odoo/blob/13.0/addons/point_of_sale/static/src/js/models.js#L622

Also I think the purpose of that function is to get **new** partners (after the pos has loaded): 
see: https://github.com/odoo/odoo/blob/13.0/addons/point_of_sale/static/src/js/models.js#L617

I do think you need to overload `prepare_new_partners_domain` to add the leaf to filter by `available_in_pos`, but you need to keep the original leafs by calling _super(), the `write_date` leaf is what gets the job done.. we shouldn't remove it https://github.com/odoo/odoo/blob/13.0/addons/point_of_sale/static/src/js/models.js#L616


That's to have the `load_new_partner` function work with this module.. But we also need to restrict the partners loaded when the pos is starting, which is actually the main purpose of this module.

To do that I think it's better to add a domain property to the `res.partner` object: https://github.com/odoo/odoo/blob/13.0/addons/point_of_sale/static/src/js/models.js#L200

I haven't tested but I think something like this should work:

```js
var PosModels = require('point_of_sale.models');

function model_load_domain(model_name, domain) {
    var models = PosModel.prototype.models;
    for (var i = 0; i < models.length; i++) {
        var model = models[i];
        if (model.model === model_name) {
            if ((model.domain instanceof Array) && model.domain.length > 0) {
                model.domain = model.fields.concat(domain || []);
            } else {
                model.domain = domain;
            }
        }
    }
}

model_load_domain("res.partner", [['available_in_pos', '=', true]]);
```